fabric-registry-sync-v0/src/main/java/net/fabricmc/fabric/impl/registry/sync/RegistrySyncManager.java

```diff
@@ -216,4 +268,67 @@ public static void unmap() throws RemapException {
 			}
 		}
 	}
+
+	public static void bootstrapRegistries() {
+		for (MutableRegistry<?> registry : Registry.REGISTRIES) {
+			if (registry instanceof ModdableRegistry) {
+				((ModdableRegistry) registry).storeIdHash(registry.getIds().hashCode());
+			}
+		}
+
+		postBootstrap = true;
+	}
+
+	private static void markModded(Registry<?> registry) {
+		if (registry instanceof ModdableRegistry) {
+			((ModdableRegistry) registry).markModded();
+		} else {
+			throw new RuntimeException("Cannot mark a none moddable registry as modded!");
+		}
+	}
+
+	public static boolean isRegistryModded(Identifier registryId) {
+		//All none minecraft registries are modded
+		if (!registryId.getNamespace().equals("minecraft")) {
+			return true;
+		}
+
+		Registry<?> registry = Registry.REGISTRIES.get(registryId);
+
+		if (registry instanceof ModdableRegistry) {
+			ModdableRegistry moddableRegistry = (ModdableRegistry) registry;
+			return moddableRegistry.isModded();
+		} else {
+			return false; //TODO what should this be?
+		}
+	}
+
+	private static class RegistryTypes {
```

Comment:
This may not be ideal, as mod provided registries should be able to define these.